x-pack/plugins/monitoring/server/routes/api/v1/logstash/pipeline.ts

```diff
@@ -66,16 +55,19 @@ export function logstashPipelineRoute(server) {
       }
       const version = getPipelineVersion(versions, pipelineHash);
 
-      // noinspection ES6MissingAwait
-      const promises = [getPipeline(req, config, clusterUuid, pipelineId, version)];
-      if (detailVertexId) {
-        promises.push(
-          getPipelineVertex(req, config, clusterUuid, pipelineId, version, detailVertexId)
-        );
-      }
+      const callGetPipelineVertexFunc = () => {
+        if (!detailVertexId) {
+          return Promise.resolve(undefined);
+        }
+
+        return getPipelineVertex(req, config, clusterUuid, pipelineId, version, detailVertexId);
+      };
 
       try {
-        const [pipeline, vertex] = await Promise.all(promises);
+        const [pipeline, vertex] = await Promise.all([
```

Comment:
This is like this so that ts is able to infer the correct type of `pipeline` and `vertex`